static/js/drafts.js

```diff
@@ -410,6 +464,15 @@ export function launch() {
                 rendered_markdown.update_elements($(this));
             });
         }
+        if (narrow_drafts.length > 0) {
+            $("#drafts-from-conversation").show();
+            if (other_drafts.length === 0) {
+                $("#other-drafts").hide();
+            }
+        } else {
+            $("#drafts-from-conversation").hide();
+            $("#other-drafts-header").hide();
+        }
```

Comment:
Let's deduplicate this logic from here and above `remove_draft` function.

Reply:
are you talking about these two lines, or more of it? it feels weird to me to abstract away the jquery `hide` / `show` calls for one branch of the if/else and not the other

Reply:
are you talking about 

```
        if (narrow_drafts.length > 0) {
            $("#drafts-from-conversation").show();
            if (other_drafts.length === 0) {
                $("#other-drafts").hide();
            }
        } else {
            $("#drafts-from-conversation").hide();
            $("#other-drafts-header").hide();
        }
```

and this?

```
    if ($("#drafts-from-conversation .draft-row").length === 0) {
        $("#drafts-from-conversation").hide();
        $("#other-drafts-header").hide();
    }
    if ($("#other-drafts .draft-row").length === 0) {
        $("#other-drafts").hide();
    }
```

So consolidating them would involve

* Using jquery queries for both instead of `narrow_drafts` and `other_drafts` for the first
* Adding logic for `$("#drafts-from-conversation").show();` to the second

It feels weird to me to do queries on elements that aren't visible, which is part of why I used jquery for hiding but variables for the initialization.

I might also be missing something, please let me know what you were thinking! I'm still learning best practices for jquery :)

Reply:
I was thinking extracting something like this:
```js
function update_rendered_drafts(has_narrow_drafts, has_other_drafts) {
    if (has_narrow_drafts) {
        // Since there are no relevant drafts from this conversation left, switch to the "all drafts" view and remove headers.
        $("#drafts-from-conversation").hide();
        $("#other-drafts-header").hide();
    } else {
        $("#drafts-from-conversation").show();
    }
    if (has_other_drafts) {
        $("#other-drafts").hide();
    }
}

```

Reply:
Ah okay I think I see what you're getting at, that's cute! thanks :)

I think it you'd need to flip some of the if statements, and I guess we can also add the no drafts logic here - what do you think?

```js
function update_rendered_drafts(has_drafts_from_conversation, has_other_drafts) {
    if (has_drafts_from_conversation) {
        $("#drafts-from-conversation").show();
    } else {
        // Since there are no relevant drafts from this conversation left, switch to the "all drafts" view and remove headers.
        $("#drafts-from-conversation").hide();
        $("#other-drafts-header").hide();
    }

    if (!has_other_drafts) {
        $("#other-drafts").hide();
    }

    if (!has_drafts_from_conversation && !has_other_drafts) {
        $("#drafts_table .no-drafts").show();
    } else {
        $("#drafts_table .no-drafts").hide();
        // Update possible dynamic elements.
        const $rendered_drafts = $drafts_table.find(
            ".message_content.rendered_markdown.restore-draft",
        );
        $rendered_drafts.each(function () {
            rendered_markdown.update_elements($(this));
        });
    }
}
```

Reply:
> I guess we can also add the no drafts logic here - what do you think?

I don't think they belong here since we will be running the whole
```
        $rendered_drafts.each(function () {
            rendered_markdown.update_elements($(this));
        });
```
everytime we remove a draft.

Reply:
Oh of course :) Okay I'll move the other piece in! Thanks